### PR TITLE
List of Participants

### PR DESCRIPTION
**Pull request by sbilodeau**

### CODE REVIEW

File: pages/reports/index.vue [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+async function createTemplateReport(templateUrl) {
+  try {
+    const source     = await ky.get(templateUrl).text();
+    const template   = Handlebars.compile(source);
+    const reprotData = await this.generateReportData();
+
+    reprotData.eventDetails = await this.getEventDetails();
+
+    _.forEach(reprotData.sections, (sections) => {
```
Comment (randyhoulahan) [inline pages/reports/index.vue:153]:
```suggestion
    _.forEach(reprotData.sections, (section) => {
```

File: pages/reports/index.vue [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+      sections.organizations = _.sortBy(sections.organizations, 'name'); // eslint-disable-line no-param-reassign
```
Comment (randyhoulahan) [inline pages/reports/index.vue:154]:
```suggestion
      section.organizations = _.sortBy(section.organizations, 'name'); // eslint-disable-line no-param-reassign
```

File: pages/reports/index.vue (hunk 1 above)
Comment (randyhoulahan) [inline pages/reports/index.vue:153]: Is _.forEach different from native for?

for( const section of reprotData.sections){ ....}

File: pages/reports/index.vue [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+  if (!_.isObject(organizationObject)) {
+    organizations.push({
+      name    : country ? this.getCountryNameByCode(country) : '',
+      acronym : curentData.organization.acronym || ' ',
+      contacts: [],
+    });
+  }
+  const organization = _.find(organizations, [ 'name', governmentName ]);
+  this.addContacts(organization.contacts, curentData);
+}
```
Comment (randyhoulahan) [inline pages/reports/index.vue:212]:
```suggestion
 function addOrganizations(organizations, currentData){
  const { organization: { government, country, acronym = '' } } = currentData;
  
  const hasCountryCode = government || country;
  const name                     = hasCountryCode ? this.getCountryNameByCode(hasCountryCode) : '';
  const orgExists               = _.find(organizations, { name });
  const newOrg                 = { name, acronym, contacts: [] };

  if(!orgExists) organizations.push(newOrg);

  const { contacts } = orgExists? orgExists : newOrg;

  this.addContacts(contacts, currentData);
}
```

File: pages/reports/index.vue [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+function addContacts(contacts, curentData) {
+  contacts.push({
+    title      : curentData.title,
+    firstName  : curentData.firstName,
+    lastName   : curentData.lastName,
+    emails     : curentData.emails.join(','),
+    designation: curentData.designation || 'Head of Biodiversity Unit',
```
Comment (randyhoulahan) [inline pages/reports/index.vue:220]: these defaults seem very specific?

File: pages/reports/index.vue [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+    }
+  }
+}
+
+async function getEventDetails() {
+  const event = await this.$kronosApi.queryEvents({ eventIds: [ '0000000052000000cbd0495c0000163c' ] });
+
+  const eventDetails  = event.records.length ? event.records[0] : {};
+  const { startDate } = eventDetails;
+  const { endDate }   = eventDetails;
```
Comment (randyhoulahan) [inline pages/reports/index.vue:187]:
```suggestion
  const { startDate, endDate } = eventDetails;
```

**Comment by randyhoulahan** [comment]
All good, just showed some alternative ways, maybe a little easier to read.  I refactored the add org function, one less look up.  I think performance is an issue here if I recall.

File: pages/reports/index.vue [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+        organizations: [],
+      },
+      {
+        title        : 'Youth',
+        organizations: [],
+      },
+    ],
+  };
+
+  while (record = await cursor.next()) { // eslint-disable-line no-cond-assign, no-await-in-loop
```
[the remark this answers is not in the document]
Comment (randyhoulahan) [reply pages/reports/index.vue:126]: I think this should/could be a const here as well.

File: pages/reports/index.vue [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+      getCountryNameByCode   : 'countries/getNameByCode',
+    }),
+  },
+  methods: {
+    addOrganizations,
+    addContacts,
+    generateReport,
+    createTemplateReport,
+    generateReportData,
+    getEventDetails,
```
Comment (sbilodeau) [inline pages/reports/index.vue:66]: utlimatly the report will be denerage from the server so you don't need anything o the scope. avoid dependin on vue

File: pages/reports/index.vue [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+
+Handlebars.registerHelper('index', (value) => value + 1);
+
+async function generateReport() {
+  this.createTemplateReport('/report-templates/cbd-list-of-participants.html');
+}
+
+async function generateReportData() {
+  const query   = {
+    eventIds                     : [ '0000000052000000cbd0495c0000163c' ],
```
Comment (sbilodeau) [inline pages/reports/index.vue:78]: this is the meetingId information you should use to display

File: pages/reports/index.vue [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+      {
+        title        : 'Academic/Research',
+        organizations: [],
+      },
+      {
+        title        : 'Youth',
+        organizations: [],
+      },
+    ],
+  };
```
Comment (sbilodeau) [inline pages/reports/index.vue:124]: section should be empty an populated using the organizationTypes passed to the query. with special handling for GOV (Parties Non-Parties)

File: pages/reports/index.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (sbilodeau) [reply pages/reports/index.vue:126]: is this working?

File: pages/reports/index.vue [hunk 10]
```diff
@@ -0,0 +1,229 @@
...
+  while (record = await cursor.next()) { // eslint-disable-line no-cond-assign, no-await-in-loop
+    records.push(record);
+    let  organizationType = null;
+    if (record.organization.organizationTypeId) organizationType = this.getOrganizationTypeById(record.organization.organizationTypeId);
+    const { sections } = reportData;
+
+    for (let index = 0; index < sections.length; index++) {
+      const section = sections[index];
+      if (section.title === organizationType.title) {
+        const { organizations } = section;
```
[the remark this answers is not in the document]
Comment (sbilodeau) [reply pages/reports/index.vue:135]: i think yes

File: pages/reports/index.vue (hunk 1 above)
Comment (sbilodeau) [reply pages/reports/index.vue:153]: the air bnb rpevent `for of` they have an explaination

File: pages/reports/index.vue (hunk 4 above)
Comment (sbilodeau) [reply pages/reports/index.vue:219]: If not designation then leave it empty

File: pages/reports/index.vue [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+    firstName  : curentData.firstName,
+    lastName   : curentData.lastName,
+    emails     : curentData.emails.join(','),
+    designation: curentData.designation || 'Head of Biodiversity Unit',
+    department : curentData.department || 'General Directorate of Environmental Policies Development',
+    phones     : curentData.phones.join(',') || '+61 2 6274 2501',
+    address    : curentData.address || 'Blvd. Zhan DArk, Norbert Jokl23',
+    city       : curentData.city || 'Yerevan',
+    country    : curentData.country ? this.getCountryNameByCode(curentData.country) : 'Armenia',
+    postalCode : curentData.postalCode || '0010',
```
Comment (sbilodeau) [inline pages/reports/index.vue:225]: drop the default values

File: pages/reports/index.vue [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+    title      : curentData.title,
+    firstName  : curentData.firstName,
+    lastName   : curentData.lastName,
+    emails     : curentData.emails.join(','),
+    designation: curentData.designation || 'Head of Biodiversity Unit',
+    department : curentData.department || 'General Directorate of Environmental Policies Development',
+    phones     : curentData.phones.join(',') || '+61 2 6274 2501',
+    address    : curentData.address || 'Blvd. Zhan DArk, Norbert Jokl23',
+    city       : curentData.city || 'Yerevan',
+    country    : curentData.country ? this.getCountryNameByCode(curentData.country) : 'Armenia',
```
Comment (sbilodeau) [inline pages/reports/index.vue:224]: keep country as is and add `countryName` instead

File: pages/reports/index.vue [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+  const organization = _.find(organizations, [ 'name', governmentName ]);
+  this.addContacts(organization.contacts, curentData);
+}
+
+function addContacts(contacts, curentData) {
+  contacts.push({
+    title      : curentData.title,
+    firstName  : curentData.firstName,
+    lastName   : curentData.lastName,
+    emails     : curentData.emails.join(','),
```
Comment (sbilodeau) [inline pages/reports/index.vue:218]: leadt the emails as is and do tje join in hnadlebars

File: pages/reports/index.vue [hunk 14]
```diff
@@ -0,0 +1,228 @@
...
+    designation: curentData.designation || 'Head of Biodiversity Unit',
+    department : curentData.department || 'General Directorate of Environmental Policies Development',
+    phones     : curentData.phones.join(',') || '+61 2 6274 2501',
+    address    : curentData.address || 'Blvd. Zhan DArk, Norbert Jokl23',
+    city       : curentData.city || 'Yerevan',
+    country    : curentData.country ? this.getCountryNameByCode(curentData.country) : 'Armenia',
+    postalCode : curentData.postalCode || '0010',
+  });
+}
```
Comment (sbilodeau) [inline pages/reports/index.vue:227]: push the whole contact as-is. 
don't override `country` field, better to add the `countryName`.
Emails and phone must remain arrays and formating is done int handlebars

File: pages/reports/index.vue [hunk 15]
```diff
@@ -0,0 +1,228 @@
...
+async function getEventDetails() {
+  const event = await this.$kronosApi.queryEvents({ eventIds: [ '0000000052000000cbd0495c0000163c' ] });
+
+  const eventDetails  = event.records.length ? event.records[0] : {};
+  const { startDate, endDate } = eventDetails;
+
+  eventDetails.startDate = toDateFormat(startDate, 'dd LLLL yyyy');
+  eventDetails.endDate   = toDateFormat(endDate, 'dd LLLL yyyy');
+  eventDetails.dateRange = `${toDateFormat(startDate, 'dd')}-${toDateFormat(endDate, 'dd')} 
+                            ${toDateFormat(endDate, 'LLLL yyyy')}`;
```
Comment (sbilodeau) [inline pages/reports/index.vue:191]: formating is done in handlerbar.
we may register 2-3 helprs '
https://handlebarsjs.com/guide/#custom-helpers eg: `dateLong` `dateRange`

Comment (sbilodeau) [reply pages/reports/index.vue:191]: Date range must check if within the same month

File: pages/reports/index.vue [hunk 16]
```diff
@@ -0,0 +1,228 @@
...
+  return eventDetails;
+}
+
+function addOrganizations(organizations, curentData) {
+  const country = curentData.organization.government || curentData.organization.country;
+
+  const governmentName = country ? this.getCountryNameByCode(country) : '';
```
Comment (sbilodeau) [inline pages/reports/index.vue:198]: an organization is considered a Goverment if organizationType is `GOV`

File: pages/reports/index.vue [hunk 17]
```diff
@@ -0,0 +1,228 @@
...
+
+  const organizationObject = _.find(organizations, [ 'name', governmentName ]);
```
Comment (sbilodeau) [inline pages/reports/index.vue:200]: refer using country code. not name

File: pages/reports/index.vue (hunk 16 above)
Comment (sbilodeau) [inline pages/reports/index.vue:198]: an organization is considered a *Goverment* if `organizationType`=`GOV` 
the we load the country from the `government` fields (NOT `country`)

File: static/report-templates/cbd-list-of-participants.html [hunk 18]
```diff
@@ -0,0 +1,433 @@
...
+		{{/with}}
+	</coverpage> <!-- COVERPAGE END -->
+	<div class="contents">
+		{{#each sections}}
+		<section>
+			<!-- Parties Section  -->
+			<header class="sectionTitle">
+				<div class="pagenext_topbar">
+					<span class="pull-left">Convention on Biological Diversity</span>
+					<span class="pull-right"><span class="uppercase">WG2020 2</span> List of Participantss</span>
```
Comment (sbilodeau) [inline static/report-templates/cbd-list-of-participants.html:396]: WG2020 2should com from meeting

File: static/report-templates/participant_report.css [hunk 19]
```diff
@@ -0,0 +1,325 @@
+body {
```
Comment (sbilodeau) [inline static/report-templates/participant_report.css:1]: delet the file if not needed anymore

File: static/report-templates/cbd-list-of-participants.html [hunk 20]
```diff
@@ -0,0 +1,433 @@
...
+				<div class="parties_box">{{title}}</div>
+			</header>
+			<content>
+				{{#each organizations}}
+				<organization>
+					<organization-name>{{name}}</organization-name>
+					{{#each contacts}}
+					<participant>
+						<span class="index">{{index @index}}</span>
+						<participant-details>
```
Comment (sbilodeau) [inline static/report-templates/cbd-list-of-participants.html:407]: I like this one!

File: static/report-templates/cbd-list-of-participants.html [hunk 21]
```diff
@@ -0,0 +1,433 @@
...
+							<div class="department">{{department}}</div>
+							<div class="address">{{address}}</div>
+							<div class="city">{{city}}</div>
+							<div class="postalCode">{{postalCode}}</div>
+							<div class="country">{{country}}</div>
+							<dl>
+								<dt class="tel">Tel:</dt>
+								<dd class="tel">{{phones}}</dd>
+								<dt class="email">Emails:</dt>
+								<dd class="email">{{emails}}</dd>
```
Comment (sbilodeau) [inline static/report-templates/cbd-list-of-participants.html:419]: Better to do the phones/emails here in the handlerbar template

File: pages/reports/index.vue [hunk 22]
```diff
@@ -0,0 +1,228 @@
...
+    registrationStatusForEventIds: [ '0000000052000000cbd0495c0000170d', '0000000052000000cbd0495c0000170f', '0000000052000000cbd0495c0000163c' ],
+    limit                        : 25,
+  };
+  const result  = await this.$kronosApi.queryContacts(query);
+  const cursor  = result.getCursor();
+  const records = [];
+  let record    = null;
+
+  const reportData = {
+    eventDetails: {},
```
Comment (sbilodeau) [inline pages/reports/index.vue:89]: rename to `event`

Comment (sbilodeau) [inline pages/reports/index.vue:89]: add the `institution` object
```get api/v2018/institutions/current```

File: pages/reports/index.vue [hunk 23]
```diff
@@ -0,0 +1,228 @@
...
+  } catch (errors) {
+    if (Array.isArray(errors)) {
+      console.log(errors);
+    } else {
+      throw errors;
+    }
+  }
+}
+
+async function getEventDetails() {
```
Comment (sbilodeau) [inline pages/reports/index.vue:182]: should pass the id from the query

File: pages/reports/index.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (JamesSCBD) [reply pages/reports/index.vue:155]: No it's not working

File: pages/reports/index.vue [hunk 24]
```diff
@@ -0,0 +1,229 @@
...
+        title        : 'Youth',
+        organizations: [],
+      },
+    ],
+  };
+
+  while (record = await cursor.next()) { // eslint-disable-line no-cond-assign, no-await-in-loop
+    records.push(record);
+    let  organizationType = null;
+    if (record.organization.organizationTypeId) organizationType = this.getOrganizationTypeById(record.organization.organizationTypeId);
```
[the remark this answers is not in the document]
Comment (JamesSCBD) [reply pages/reports/index.vue]: Done!

File: pages/reports/index.vue (hunk 7 above)
Comment (JamesSCBD) [reply pages/reports/index.vue:66]: I don't understand. Are you want to remove this methods? Can you brief me more.

File: pages/reports/index.vue (hunk 9 above)
Comment (JamesSCBD) [reply pages/reports/index.vue]: I am working on it

**Comment by JamesSCBD** [comment]
Hi @sbilodeau 

All issue is resolved except this one 
>section should be empty an populated using the organizationTypes passed to the query. with special handling for GOV (Parties Non-Parties)

also added a comment on this for explain more 
>utlimatly the report will be denerage from the server so you don't need anything o the scope. avoid dependin on vue

File: pages/reports/index.vue (hunk 7 above)
Comment (sbilodeau) [reply pages/reports/index.vue:66]: The report will be generated server side probably not in vue. You the functions on under the components context “this” but they should be stand alone (not using this.addContact)

Don’t worry, it’s not important for now

File: pages/reports/index.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (sbilodeau) [reply pages/reports/index.vue:155]: then we delete the suggestion!

File: filters/datetime.js [hunk 25]
```diff
@@ -23,4 +23,6 @@ function toDateFormat(value, format) {
   return dateValue ? dateValue.toFormat(format) : '';
 }
 
-export { datetime, date, time };
+export {
+  datetime, date, time, toDateFormat,
```
Comment (sbilodeau) [inline filters/datetime.js:27]: can you rename `toDateFormat` => `formatData`

File: pages/reports/index.vue [hunk 26]
```diff
@@ -0,0 +1,269 @@
...
+  hb.registerHelper('date', () => {
+    const d    = new Date();
+    let month  = `${d.getMonth() + 1}`;
+    let day    = `${d.getDate()}`;
+    const year = d.getFullYear();
+
+    if (month.length < 2) month = `0${month}`;
+    if (day.length < 2) day = `0${day}`;
+
+    return [ year, month, day ].join('-');
```
Comment (sbilodeau) [inline pages/reports/index.vue:232]: very complex. why not just using formatDate from the filters?

File: pages/reports/index.vue [hunk 27]
```diff
@@ -0,0 +1,269 @@
...
+  });
+
+  hb.registerHelper('dateFormat', (value) => toDateFormat(value, 'dd LLLL yyyy'));
```
Comment (sbilodeau) [inline pages/reports/index.vue:235]: `dateFormat` have no meaning. helper must be called `longDate`

File: pages/reports/index.vue [hunk 28]
```diff
@@ -0,0 +1,269 @@
...
+
+  hb.registerHelper('dateRange', (startDate, endDate) => {
+    startDate = new Date(startDate); // eslint-disable-line no-param-reassign
+    endDate   = new Date(endDate); // eslint-disable-line no-param-reassign
```
Comment (sbilodeau) [inline pages/reports/index.vue:239]: avoid reassinged here

File: pages/reports/index.vue [hunk 29]
```diff
@@ -0,0 +1,269 @@
...
+    if (startDate.getMonth() === endDate.getMonth()) {
+      return `${toDateFormat(startDate, 'dd')}-${toDateFormat(endDate, 'dd')} ${toDateFormat(endDate, 'LLLL yyyy')}`;
```
Comment (sbilodeau) [inline pages/reports/index.vue:241]: what not groups the enddate for into one?

```suggestion
      return `${toDateFormat(startDate, 'dd')}-${toDateFormat(endDate, 'dd LLLL yyyy')}`;
```

File: pages/reports/index.vue [hunk 30]
```diff
@@ -0,0 +1,269 @@
...
+    }
+    return `${toDateFormat(startDate, 'dd LLLL yyyy')} - ${toDateFormat(endDate, 'dd LLLL yyyy')}`;
```
Comment (sbilodeau) [inline pages/reports/index.vue:243]: if year is the same no need to repeat

File: pages/reports/index.vue [hunk 31]
```diff
@@ -0,0 +1,269 @@
...
+
+    if (month.length < 2) month = `0${month}`;
+    if (day.length < 2) day = `0${day}`;
+
+    return [ year, month, day ].join('-');
+  });
+
+  hb.registerHelper('dateFormat', (value) => toDateFormat(value, 'dd LLLL yyyy'));
+
+  hb.registerHelper('dateRange', (startDate, endDate) => {
```
Comment (sbilodeau) [inline pages/reports/index.vue:237]: rename to `longDateRange` to be consistent with `longDate`

File: static/report-templates/cbd-list-of-participants.html [hunk 32]
```diff
@@ -0,0 +1,447 @@
...
+			font-size: 20px;
+		}
+
+		h4.participants_date {
+			float: left;
+			width: 100%;
+			text-align: center;
+			margin: 150px 0 50px;
+			font-size: 23px;
+		}
```
Comment (sbilodeau) [inline static/report-templates/cbd-list-of-participants.html:160]: simplify, scope and clean the css.

File: static/report-templates/cbd-list-of-participants.html [hunk 33]
```diff
@@ -0,0 +1,447 @@
...
+
+			@bottom-right {
+				content: none;
+			}
+		}
+	</style>
+</head>
+
+<body>
+	<div class="footerDate">{{date ''}}</div>
```
Comment (sbilodeau) [inline static/report-templates/cbd-list-of-participants.html:353]: I don't undertand why this is there and what this doing. if it's the footer better to put near the end of the page

File: static/report-templates/cbd-list-of-participants.html [hunk 34]
```diff
@@ -0,0 +1,447 @@
...
+			</div>
+			<div class="coverhead-botttombar">
+				<h1 class="logo"><img src="mailLogo.png"></h1>
+				<div class="header_rightSec">
+					<p>Distr.<br><span class="uppercase">GENERAL</span></p>
+					<p><span class="uppercase">{{code}}</span><br>{{dateFormat startDate}}</p>
+					<p class="uppercase">ENGLISH ONLY</p>
+				</div>
+			</div>
+		</div>
```
Comment (sbilodeau) [inline static/report-templates/cbd-list-of-participants.html:369]: put converpage header in `<header>`

Comment (sbilodeau) [reply static/report-templates/cbd-list-of-participants.html:369]: try to use always the same pattern:
```html
<header/>
<content/>
<footer/>
```
you may have to put the footer before the content

File: static/report-templates/cbd-list-of-participants.html [hunk 35]
```diff
@@ -0,0 +1,447 @@
...
+							{{#if phones}}
+								<dt class="tel">Tel.:</dt>
+								<dd class="tel">
+									{{#each phones}}
+										<a href="tel:{{this}}">{{this}}</a>
+									{{/each}}
+									</dd>
+							{{/if}}
+							{{#if emails}}
+								<dt class="email">Email:</dt>
```
Comment (sbilodeau) [inline static/report-templates/cbd-list-of-participants.html:428]: weneed to add fax

File: static/report-templates/cbd-list-of-participants.html [hunk 36]
```diff
@@ -0,0 +1,447 @@
...
+								<dd class="email">
+									{{#each emails}}
```
Comment (sbilodeau) [inline static/report-templates/cbd-list-of-participants.html:430]: if you compare with the original, many have more than one addresses and this is only showining 1. 
maybe the old one include emailCCs? can you have a look  and fix?

File: filters/datetime.js (hunk 25 above)
Comment (JamesSCBD) [reply filters/datetime.js]: Done!

File: pages/reports/index.vue (hunk 26 above)
Comment (JamesSCBD) [reply pages/reports/index.vue]: Done! Use `generatedOn` variable

File: pages/reports/index.vue (hunk 27 above)
Comment (JamesSCBD) [reply pages/reports/index.vue]: Done!

File: pages/reports/index.vue (hunk 28 above)
Comment (JamesSCBD) [reply pages/reports/index.vue]: Done

File: pages/reports/index.vue (hunk 29 above)
Comment (JamesSCBD) [reply pages/reports/index.vue]: Done

File: pages/reports/index.vue (hunk 30 above)
Comment (JamesSCBD) [reply pages/reports/index.vue]: Done

File: pages/reports/index.vue (hunk 31 above)
Comment (JamesSCBD) [reply pages/reports/index.vue]: Done

File: static/report-templates/cbd-list-of-participants.html (hunk 32 above)
Comment (JamesSCBD) [reply static/report-templates/cbd-list-of-participants.html]: Done but i didn't add any css right now

File: static/report-templates/cbd-list-of-participants.html (hunk 33 above)
Comment (JamesSCBD) [reply static/report-templates/cbd-list-of-participants.html]: This is used to show date on left  footer. I try to put this div in other place but didn't work. If you have any suggestion how to solve this issue let me know.

File: static/report-templates/cbd-list-of-participants.html (hunk 34 above)
Comment (JamesSCBD) [reply static/report-templates/cbd-list-of-participants.html]: Done

File: static/report-templates/cbd-list-of-participants.html [hunk 37]
```diff
@@ -0,0 +1,418 @@
...
+		}
+
+		/** participant css**/
+		section>content>organization>participant {
+			display: block;
+		}	
+		section>content>organization>participant {
+			margin-bottom: 16px;
+			position: relative;
+			padding-left: 37px;
```
Comment (sbilodeau) [inline static/report-templates/cbd-list-of-participants.html:154]: better to use pt than px when working with printed media. 
1 pt = 1/72 inche
1 px= 1 screen px => ~ 1/96 inche 

```
pt = 0.75 * px
```

**Pull request merged**